game/src/hud.js

```diff
@@ -14,19 +16,38 @@ var createHud = function() {
     scoreText.width = .5;
     scoreText.height = .15;
 
+    // Create a High Score Text Block
+    highScoreText2 = new BABYLON.GUI.TextBlock();
+    highScoreText2.fontFamily = "Helvetica, sans-serif";
+    highScoreText2.color = "white";
+    highScoreText2.fontSize = 48;
+    highScoreText2.verticalAlignment = BABYLON.GUI.TextBlock.VERTICAL_ALIGNMENT_TOP;
+    highScoreText2.horizontalAlignment = BABYLON.GUI.TextBlock.HORIZONTAL_ALIGNMENT_LEFT;
+    highScoreText2.width = .5;
+    highScoreText2.height = .15;
+    
     updateScoreText();
 
     hudTexture.addControl(scoreText);
+    
 }
 
 var updateScoreText = function() {
     scoreText.text = "Score: " + score;
+    if(score > highScore){
+        highScore = score;
+    }
+    highScoreText2.text = "High Score: " + highScore;
+    
+    hudTexture.addControl(highScoreText2);
```

Comment:
`hudTexture` is only defined in the function above so it's out of scope here and not defined.

You should be able to move the `var hudTexture` line to define it globally for now to fix this (line 7).

Reply:
Taking another look at this. I think this line to add (and the line below to remove) the control could be removed and this `addControl` might fit on line 32 when the controls are added to the scene.

Right now we might get multiple invocations of the "addControl" that do not align with the "removeControl" invocations.